src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -216,27 +225,22 @@ class boss_gormok : public CreatureScript
             void EnterCombat(Unit* /*who*/) override
             {
                 _EnterCombat();
-                me->SetInCombatWithZone();
                 instance->SetData(TYPE_NORTHREND_BEASTS, GORMOK_IN_PROGRESS);
-
-                for (uint8 i = 0; i < MAX_SNOBOLDS; i++)
-                {
-                    if (Creature* pSnobold = DoSpawnCreature(NPC_SNOBOLD_VASSAL, 0, 0, 0, 0, TEMPSUMMON_CORPSE_DESPAWN, 0))
-                    {
-                        pSnobold->EnterVehicle(me, i);
-                        pSnobold->SetInCombatWithZone();
-                        pSnobold->AI()->DoAction(ACTION_ENABLE_FIRE_BOMB);
-                    }
-                }
             }
 
             void DamageTaken(Unit* /*who*/, uint32& damage) override
             {
                 // despawn the remaining passengers on death
                 if (damage >= me->GetHealth())
                     for (uint8 i = 0; i < MAX_SNOBOLDS; ++i)
-                        if (Unit* pSnobold = me->GetVehicleKit()->GetPassenger(i))
-                            pSnobold->ToCreature()->DespawnOrUnsummon();
+                        if (Unit* snobold = me->GetVehicleKit()->GetPassenger(i))
+                            snobold->ToCreature()->DespawnOrUnsummon();
```

Comment:
Because he should despawn only creatures that still in your back.
Snobolds in players back is not despawned